checker/jdk/index/src/java/nio/CharBuffer.java

```diff
@@ -0,0 +1,1523 @@
+     * (inclusive) and the limit (exclusive); that is, it is equivalent to
+     * <tt>remaining()</tt>. </p>
+     *
+     * @return  The length of this character buffer
+     */
+    @SuppressWarnings("index:return.type.incompatible") // remaining is equivalent to the length of the character buffer as explained in the documentation
+    public final @NonNegative @LTEqLengthOf("this") int length() {
+        return remaining();
+    }
+
+    /**
+     * Reads the character at the given index relative to the current
+     * position.
+     *
+     * @param  index
+     *         The index of the character to be read, relative to the position;
+     *         must be non-negative and smaller than <tt>remaining()</tt>
+     *
+     * @return  The character at index
+     *          <tt>position()&nbsp;+&nbsp;index</tt>
+     *
+     * @throws  IndexOutOfBoundsException
+     *          If the preconditions on <tt>index</tt> do not hold
+     */
+    @SuppressWarnings({"index:override.param.invalid", "index:argument.type.incompatible"}) /*
+    index has to be @LTLengthOf("this.remaining()") as written in the documentation.
+    #1: index being @LTLengthOf("this.remaining()") => position() + checkIndex(index, 1) is @LTLengthOf("this.limit")
+    */
+    public final char charAt(@NonNegative @LTLengthOf("this.remaining()") int index) {
+        return get(position() + checkIndex(index, 1));
+    }
+
+    /**
+     * Creates a new character buffer that represents the specified subsequence
+     * of this buffer, relative to the current position.
+     *
+     * <p> The new buffer will share this buffer's content; that is, if the
+     * content of this buffer is mutable then modifications to one buffer will
+     * cause the other to be modified.  The new buffer's capacity will be that
+     * of this buffer, its position will be
+     * <tt>position()</tt>&nbsp;+&nbsp;<tt>start</tt>, and its limit will be
+     * <tt>position()</tt>&nbsp;+&nbsp;<tt>end</tt>.  The new buffer will be
+     * direct if, and only if, this buffer is direct, and it will be read-only
+     * if, and only if, this buffer is read-only.  </p>
+     *
+     * @param  start
+     *         The index, relative to the current position, of the first
+     *         character in the subsequence; must be non-negative and no larger
+     *         than <tt>remaining()</tt>
+     *
+     * @param  end
+     *         The index, relative to the current position, of the character
+     *         following the last character in the subsequence; must be no
+     *         smaller than <tt>start</tt> and no larger than
+     *         <tt>remaining()</tt>
+     *
+     * @return  The new character buffer
+     *
+     * @throws  IndexOutOfBoundsException
+     *          If the preconditions on <tt>start</tt> and <tt>end</tt>
+     *          do not hold
+     */
+    @SuppressWarnings("index:override.param.invalid") // the annotations are as per the documentation
```

Comment:
I don't think you should override the annotations on the super class. If they're wrong, we should change them.

However, I think these annotations are wrong: `@LTEqLengthOf` doesn't make sense when applied to `"this.remaining()"`, which is an `int`, not an array.